src/databricks/labs/remorph/reconcile/query_builder_refactored/expression_generator.py

```diff
@@ -0,0 +1,124 @@
+from collections.abc import Callable
+from functools import partial
+
+from sqlglot import expressions as exp
+from sqlglot.expressions import (
+    Alias,
+    Anonymous,
+    Coalesce,
+    Column,
+    DataType,
+    Expression,
+    From,
+    Join,
+    JSONFormat,
+    Literal,
+    Trim,
+)
+
+
+def coalesce(expr: Expression, default="0", is_string=False) -> Coalesce | Expression:
+    level = 0 if isinstance(expr, exp.Column) else 1
+    new_expr = expr.copy()
+    for node in new_expr.dfs():
+        if isinstance(node, exp.Column):
+            column_name = node.name
+            table_name = node.table
+            func = exp.Coalesce(
+                this=exp.Column(this=column_name, table=table_name),
+                expressions=[exp.Literal(this=default, is_string=is_string)],
+            )
+            if level == 0:
+                return func
+            node.replace(func)
+    return new_expr
+
+
+def trim(expr: Expression) -> Trim | Expression:
+    level = 0 if isinstance(expr, exp.Column) else 1
+    new_expr = expr.copy()
+    for node in new_expr.dfs():
+        if isinstance(node, exp.Column):
+            column_name = node.name
+            table_name = node.table
+            func = exp.Trim(this=exp.Column(this=column_name, table=table_name))
+            if level == 0:
+                return func
+            node.replace(func)
+    return new_expr
+
+
+def json_format(expr: Expression, options: dict[str, str] | None = None) -> JSONFormat | Expression:
+    level = 0 if isinstance(expr, exp.Column) else 1
+    new_expr = expr.copy()
+    for node in new_expr.dfs():
+        if isinstance(node, exp.Column):
+            column_name = node.name
+            table_name = node.table
+            func = exp.JSONFormat(this=exp.Column(this=column_name, table=table_name), options=options)
+            if level == 0:
+                return func
+            node.replace(func)
+    return new_expr
+
+
+def anonymous(node, func: str) -> Anonymous | Expression:
+    if isinstance(node, exp.Column):
+        column_name = node.name
+        table_name = node.table
+        return exp.Anonymous(this=func, expressions=[exp.Column(this=column_name, table=table_name)])
+    return node
+
+
+def build_column(this, table_name="", quoted=False) -> Column:
+    return exp.Column(this=exp.Identifier(this=this, quoted=quoted), table=table_name)
+
+
+def build_literal(this, is_string=True) -> Literal:
+    return exp.Literal(this=this, is_string=is_string)
+
+
+def build_literal_alias(this: exp.ExpOrStr, alias="", quoted=False, is_string=True) -> Alias:
```

Comment:
redundant can be handled inside `build_literal`

Reply:
Done.